apps/initializer/lib/initializer.ex

```diff
@@ -49,6 +55,18 @@ defmodule Initializer do
         end
       end
 
+      @retry with: constant_backoff(100) |> take(10)
+      defp do_on_start(state) do
+        case on_start(state) do
+          {:ok, new_state} ->
+            {:ok, new_state}
+
+          {:error, reason} ->
+            {:error,
+             "on_start for application init failed, retrying reading Brook state from store..."}
```

Comment:
i would make this error message more generic; might not necessarily be trying to initialize from Brook state.

Reply:
i would just change the error case to:
```
  error ->
    error
```
to let the underlying app's implementation of it's error message bubble up.

Reply:
Fixed. Good call.